service/lib/agama/progress.rb

```diff
@@ -154,5 +188,11 @@ def to_s
 
       "#{current_step.description} (#{@counter}/#{total_steps})"
     end
+
+  private
+
+    def description_for(step)
+      @descriptions[step - 1] || format(_("Step %s/%s"), step, total_steps)
```

Comment:
NP: I would not generate a default description. If no description is known for the step (e.g., `Propress.with_size` and `progress.step {}`), then I would create a `Step` with empty description.

Note that the steps counter is generated by the clients if needed (e.g., "Creating partitions (2/4)") from the info on D-Bus (`CurrentStep` and `TotalSteps`). With the default step description proposed here, the clients have no clue whether the counter is needed or it already comes in the description.